compiler/qsc_parse/src/lex/cooked.rs

```diff
@@ -365,7 +376,26 @@ impl<'a> Lexer<'a> {
                 self.expect(Single::Amp, TokenKind::ClosedBinOp(op))?;
                 Ok(self.closed_bin_op(op))
             }
-            Single::Apos => Ok(TokenKind::Apos),
+            Single::Apos => {
+                let tok = self.tokens.next();
+                match tok {
+                    Some(raw::Token {
+                        kind: raw::TokenKind::Ident,
+                        ..
+                    }) => Ok(TokenKind::AposIdent),
+                    Some(raw::Token { offset, kind }) => {
+                        let mut tokens = self.tokens.clone();
+                        let hi = tokens.nth(1).map_or_else(|| self.len, |t| t.offset);
+                        let span = Span { lo: offset, hi };
+                        Err(Error::IncompleteAposIdent(kind, span))
+                    }
+                    None => {
+                        let lo = self.len;
+                        let span = Span { lo, hi: lo };
+                        Err(Error::IncompleteAposIdentEof(span))
+                    }
+                }
+            }
```

Comment:
I don't think we should just copy/paste this from Lexer::expect - handling any Token is more general than handling only Single tokens, so there should be a way to refactor and share the logic.